File: crates/selene-resolve/tests/fixtures/dispatch/laravel/app/Http/Controllers/ArticleController.php
<?php

namespace App\Http\Controllers;

use App\Services\ArticleService;

class ArticleController
{
    private ArticleService $articleService;

    public function index()
    {
        return $this->articleService->listArticles();
    }

    public function store()
    {
        return $this->articleService->createArticle();
    }
}

File: crates/selene-resolve/tests/fixtures/dispatch/spring/src/main/java/com/example/controller/ArticleController.java
package com.example.controller;

import com.example.service.ArticleService;

@RestController
@RequestMapping("/articles")
public class ArticleController {

    private final ArticleService articleService;

    @Value("${app.greeting}")
    private String greeting;

    @GetMapping
    public String getAll() {
        return articleService.listArticles();
    }
}
